src/engine/source/kvdb/src/kvdb.cpp

```diff
@@ -1,561 +1,529 @@
- * @return int position in vector of CFs
+ * @return true succesfully destructed
+ * @return false unsuccesfully destructed
  */
-int CFIndexInAvailableArray(std::string const &column_family_name) {
-    for (int i = 0; i < column_families.size() ; i++ ) {
-        if(!column_family_name.compare(column_families.at(i).name))
-        {
-            return i;
-        }
+bool KVDB::destroy()
+{
+    close();
+    rocksdb::Status s =
+        rocksdb::DestroyDB(m_path, rocksdb::Options(), CFDescriptors);
+    if (!s.ok())
+    {
+        WAZUH_LOG_ERROR("couldn't destroy db, error: [{}]", s.ToString());
+        m_state = State::Error;
+        return false;
     }
-    return 0;
+    return true;
 }
 
 /**
- * @brief Create a Column Family object
+ * @brief write a value from to a key inside a CF without value copying
  *
- * @param column_family_name std::string if it's no part of column_families_available
- * it will be added to it.
- * @return true successfull creation of FC in DB or FC alredy in DB
- * @return false unsuccessfull creation of FC in DB
+ * @param key where to write the value
+ * @param value the value that will be writen
+ * @param columnName where to write the key
+ * @return true If the proccess finished successfully
+ * @return false If the proccess didn't finished successfully
  */
-bool CreateColumnFamily(std::string const column_family_name) {
-    DB *db;
-    Status s;
-    std::vector<ColumnFamilyHandle*> handlers;
-    DBOptions dbOptions;
-    std::vector<std::string> column_families_available;
-    bool result = true;
-
-    if(column_family_name.empty()) {
-        LOG(ERROR) << "[" << __func__ << "]" << " cant create a family column without name " << std::endl;
+bool KVDB::write(const std::string &key,
+                 const std::string &value,
+                 const std::string &columnName)
+{
+    CFHMap::const_iterator cfh = m_CFHandlesMap.find(columnName);
+    if (cfh == m_CFHandlesMap.end())
+    {
+        WAZUH_LOG_ERROR("couldn't write to DB unknown column name");
         return false;
     }
 
-    dbOptions.IncreaseParallelism();
-    dbOptions.create_if_missing = true;
-    dbOptions.error_if_exists = false;
-    dbOptions.create_missing_column_families = true;
-
-    UpdateColumnFamiliesList();
+    if (m_state == State::Open)
+    {
+        rocksdb::Status s = m_db->Put(kOptions.write,
+                                      cfh->second,
+                                      rocksdb::Slice(key),
+                                      rocksdb::Slice(value));
+        if (!s.ok())
+        {
+            WAZUH_LOG_ERROR("couldn't insert value into CF, error: [{}]",
+                            s.ToString());
+            return false;
+        }
 
-    if(CFIndexInAvailableArray(column_family_name)) {
-        LOG(INFO) << "[" << __func__ << "]" << " cant create a family column already present" << std::endl;
+        WAZUH_LOG_INFO("value insertion OK [{},{}] into CF name : [{}]",
```

Comment:
this should probably be a DEBUG log instead

Reply:
done